content/blog/jamstack-with-pulumi/index.md

```diff
@@ -0,0 +1,359 @@
+
+In the container command line, we will use hugo to create a new site called *platypus*.
+
+```bash
+$ hugo new site platypus
+$ cd platypus
+```
+
+Next, we’ll customize the site with a theme. There are many [themes available](https://themes.gohugo.io/), but for this example, we'll use the *ananke* theme. Download the theme to the container, unzip it, and rename the unzipped directory to *ananke*.
+
+```bash
+$ wget https://github.com/budparr/gohugo-theme-ananke/archive/master.zip
+$ unzip master.zip
+$ cp -rf gohugo-theme-ananke-master ./themes/ananke
+$ rm -rf master.zip gohugo-theme-ananke-master/
+```
+
+Configure the site by editing the *config.toml* file with the following.
+
+```toml
+baseURL = "http://www.<your_domain>.com/"
+languageCode = "en-us"
+title = "Platypus"
+theme = "ananke"
+```
+
+Use hugo to create a new markdown page in the posts directory.
+
+```bash
+$ hugo new posts/platypus.md
+```
+
+Now it’s time to add content to your site. You can use your favorite editor to write and edit content in your working directory on your machine instead of using the container shell. Edit the markdown file, below is an example page but feel free to add your content.
+
+```md
+## Platypus of the Day
+
+**This platypus is a cool critter!**
+
+![platypus](https://static01.nyt.com/images/2017/08/01/science/29TB-PLATYPUS1/29TB-PLATYPUS1-superJumbo.jpg)
+```
+
+Finally, we can build the static site with Hugo in the container command line.
+
+```bash
+$ hugo -D
+```
+
+Hugo will create a *public* directory with the static website.
+
+## Building the Cloud Infrastructure
+
+Configuring a static website can be complicated when using either the AWS web interfaces or the AWS CLI. To deploy a static website, we need to create and configure the following cloud resources.
+
+- [Amazon S3](https://aws.amazon.com/s3/) to store the website's contents
+- [Amazon CloudFront](https://aws.amazon.com/cloudfront/) for the CDN serving content
+- [Amazon Route53](https://aws.amazon.com/route53/) to set up the DNS for the website
+- [Amazon Certificate Manager](https://aws.amazon.com/certificate-manager/) for securing the site via HTTPS
+
+Using Pulumi to deploy Infrastructure as Code, we can create and build our infrastructure using Python or your favorite programming language. If you haven't installed Pulumi and configured it to work with your AWS credentials, follow the [Getting Started with AWS guide]({{< relref "/docs/get-started/aws">}}).
+
+To get started building our infrastructure, we’ll download the Python example for setting up a static website with AWS. There are many [examples on our GitHub repository](https://github.com/pulumi/), but we can clone just the [AWS Python Static Website example](https://github.com/pulumi/examples/tree/master/aws-py-static-website) using a sparse clone:
+
+```bash
+$ mkdir aws-website
```

Comment:
We should probably mention where this `mkdir` should be run, in relation to the Hugo site root, and that (presumably) we're no longer in the container here.